bionic/aip/task.py

```diff
@@ -0,0 +1,156 @@
+""" Data model for task running on AI platform
+"""
+import logging
+import os
+import sys
+from dataclasses import dataclass
+from typing import Callable, Optional
+
+import blocks
+import cloudpickle
+import googleapiclient.discovery
+from bionic.aip.future import Future
+from bionic.aip.tune import Tune
+
+
+@dataclass(frozen=True)
+class Resource:
+    """Description of a computing resource, which can be a single machine or a cluster of machines
+
+    In the future could be extended to support GPUs and similar
+    """
+
+    machine: str
+    worker_count: Optional[int] = None
+    worker_machine: Optional[str] = None
+
+
+@dataclass(frozen=True)
+class Job:
+    """A job groups together individual tasks, all of which share these properties
+
+    Attributes
+    ----------
+    uuid: str
+        The globally unique job name on AI platform
+    image: str
+        The name of the docker image for the tasks
+    project: str
+        The GCP project where the jobs will be run
+
+    """
+
+    uuid: str
+    image: str
+    project: str
+    account: str = None
+    network: str = None
+
+    def __post_init__(self):
+        if "gcr.io" not in self.image:
+            # NOTE this is just for convenient image shorthand
+            object.__setattr__(self, "image", f"gcr.io/{self.project}/{self.image}")
+
+
+@dataclass(frozen=True)
+class Task:
+    name: str
+    job: Job
+    function: Callable
+    resource: Resource
+    tune: Optional[Tune] = None
+
+    def job_id(self):
+        return f"{self.job.uuid}_{self.name}"
+
+    def path_inputs(self):
+        # TODO bionic will need to customize
+        return f"gs://{self.job.project}/bionic/{self.job.uuid}/{self.name}-inputs.pkl"
```

Comment:
I think it is fine to leave this as it is. I can see if we should convert them into a bionic core entity and pass it to `Task`. But what do you think about rewriting the comment since it is already in Bionic? Maybe something like "Pass the input and output paths to the class".

Reply:
i'll clarify this comment, what i mean is that we probably want to put these inside a gcs cache location if it is available on the flow, and additional that the filetype might not always be pickle if we integrate serializers.